app/services/publisher_balance_getter.rb

```diff
@@ -0,0 +1,78 @@
+# Used by the PublisherWalletGetter to retrieve balances
+class PublisherBalanceGetter < BaseApiClient
+  attr_reader :publisher
+
+  def initialize(publisher:)
+    @publisher = publisher
+  end
+
+  def perform
+    return [] if publisher.channels.verified.empty?
+    return perform_offline if Rails.application.secrets[:api_eyeshade_offline]
+
+    channel_balances_response = connection.get do |request|
+      request.headers["Authorization"] = api_authorization_header
+      request.url("/v1/balances?account=#{URI.escape(publisher.owner_identifier)}#{URI.escape(channels_query_string)}")
+    end
+
+    channel_balances = JSON.parse(channel_balances_response.body)
+
+    # Eyeshade returns an empty response for verified channels
+    # with no balance, so we must fill in 0.00 balances
+
+    # Find verified channel IDs with no balance
+    verified_channel_ids = []
+    publisher.channels.verified.each {|channel| verified_channel_ids.push(channel.details.channel_identifier)}
+
+    channel_balances.each do |channel_balance|
+      channel_identifier = channel_balance["account"]
+      verified_channel_ids.delete(channel_identifier) if verified_channel_ids.include?(channel_identifier)
+    end
+
+    # Set balances to 0.00
+    verified_channel_ids.each do |id|
```

Comment:
You could replace the block by doing

verified_channel_ids.map do |verified_channel_id| 'account' => '#{verified_channel_id}', 'balance' => "0.00" } end

Reply:
I'm not sure a `.map` makes sense here since we are pushing each result of the loop into `channel_balances`, which is an array that already has contents prior to the loop.

Reply:
`channel_balances += verified_channel_ids.map { |verified_channel_id| { 'account' => '#{verified_channel_id}', 'balance' => "0.00" } }`